fingerprint2.js

```diff
@@ -822,6 +832,24 @@
       var elem = document.createElement("canvas");
       return !!(elem.getContext && elem.getContext("2d"));
     },
+    isWebGlSupported: function() {
+      if (!this.isCanvasSupported() || !window.WebGLRenderingContext) {
+        return false;
+      }
+
+      var canvas = document.createElement("canvas"),
+          glContext = canvas.getContext && (canvas.getContext("webgl") || canvas.getContext("experimental-webgl"));
+
+      if (glContext && !glContext.getShaderPrecisionFormat) {
+        if (typeof NODEBUG === "undefined") {
+          this.log("This browser supports webgl, but lacks some functionality (getShaderPrecisionFormat not avaialable)");
```

Comment:
I think we should still do WebGL FP, even if getShaderPrecisionFormat is not available. We should just skip this part in the WebGL code for those devices. Please add a separate issue for this, and remove this check.
I'll implement the fix for Galaxy S2 right now.